### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by umarcor**
This PR brings several updates to CI:

- It allows running the workflow for any branch or PR.
- Fail-fast is set to false (defaults to true).
- The style is updated and symbols/icons are added.
- Python 3.10 is added.
- CRON and workflow_dispatch events are added/enabled.

macos and windows jobs are temporarily disabled because Edalize seems to fail on those platforms. The workflow in branch `master` is currently green because the system is always Ubuntu (the matrix is ignored): see https://github.com/olofk/edalize/blob/master/.github/workflows/ci.yml#L12.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,29 +2,43 @@ name: CI
 
 on:
   push:
-    branches: [ master ]
   pull_request:
-    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * 5'
+  workflow_dispatch:
```
Comment (imphil) [inline .github/workflows/ci.yml:8]: Can you explain the motivation for these changes?

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,10 @@
+[tool.tox]
```
Comment (imphil) [inline pyproject.toml:1]: Can you explain why this change is beneficial? Embedding an ini file as string in another ini file doesn't look like an improvement to me.

**Comment by imphil** [comment]
Thanks for your pull request @umarcor! I'd like to understand the motivation for some of the changes better, can you have a look at the comments? Thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (umarcor) [reply .github/workflows/ci.yml:8]: The scheduled/CRON event ensures that CI is executed once a week, despite the codebase not being updated. CI did not run in the last 11 days, since the last commit was pushed until I opened a PR. Since edalize has dependencies, it is desirable to run CI periodically in order to catch regressions/breaking changes.

The workflow_dispatch event allows triggering the workflow manually through the web GUI, or through the GitHub API.

File: pyproject.toml (hunk 2 above)
Comment (umarcor) [reply pyproject.toml:1]: `pyproject.toml` is a configuration file used by multiple projects, instead of having one configuration file for each. Particularly, apart from `tox`, I'd like to propose a PR for using `black`. Both `tox` and `black` support `pyproject.toml`. See, for instance, VUnit (https://github.com/VUnit/vunit/blob/master/pyproject.toml). See also https://github.com/carlosperate/awesome-pyproject.

Comment (imphil) [reply pyproject.toml:1]: I would agree if tox would "natively" read pyproject.toml. But embedding ini within an ini is a step backwards IMO. I'd ask you to revert that. Having a different configuration file for a different tool isn't a problem in any way.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (imphil) [reply .github/workflows/ci.yml:8]: Can you add a comment to that effect to the file?

Comment (umarcor) [reply .github/workflows/ci.yml]: Done.

File: pyproject.toml (hunk 2 above)
Comment (umarcor) [reply pyproject.toml]: Done.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -2,29 +2,49 @@ name: CI
 
 on:
   push:
-    branches: [ master ]
   pull_request:
-    branches: [ master ]
+  schedule:
```
Comment (imphil) [inline .github/workflows/ci.yml:6]: Let's make those comments a bit more helpful and explain *why* the constructs are used.

```suggestion
  # Run CI daily even without code changes to ensure tests pass with 
  # updated dependencies.
  schedule:
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -2,29 +2,49 @@ name: CI
...
+    - cron: '0 0 * * 5'
+  # See:
+  # * https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#scheduled-events
+  # * https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07
+  workflow_dispatch:
```
Comment (imphil) [inline .github/workflows/ci.yml:11]:
```suggestion
  # Allow triggering a CI run from the web UI.
  workflow_dispatch:
```

**Review by imphil — COMMENTED** [review]
We're almost there with two comments to go.

And you can fold the "[CI] Add refs to documentation about 'schedule' and 'workflow_dispatc" commit into the "[CI] Add CRON and workflow_dispatch events" commit.

**Comment by umarcor** [comment]
Just squashed and force-pushed. I successful, I think it is ready to merge.

**Review by imphil — APPROVED** [review]
Thanks!
